pyfar/classes/audio.py

```diff
@@ -25,23 +25,33 @@ class _Audio():
     # (e.g. __rmul__)
     __array_priority__ = 1.0
 
-    def __init__(self, domain, comment=None, dtype=np.double):
+    def __init__(self, data, domain, comment=None, dtype=None):
 
         # initialize valid parameter spaces
         self._VALID_DOMAINS = ["time", "freq"]
 
         # initialize global parameters
         self.comment = comment
-        self._dtype = dtype
         if domain in self._VALID_DOMAINS:
             self._domain = domain
         else:
             raise ValueError("Incorrect domain, needs to be time/freq.")
 
-        # initialize data with nan (this should make clear that this is an
-        # abstract private class that does not hold data and prevent class
-        # methods to fail that require data.)
-        self._data = np.atleast_2d(np.nan)
+        # set data and check dtype
+        if dtype not in (None, float, complex):
+            raise ValueError("dtype must be None, float, or complex")
+        elif dtype is None:
+            self._data = np.atleast_2d(np.asarray(data))
+
+            # convert ints to float and check dtype of input
+            if str(self._data.dtype).startswith("int"):
+                self._data = self._data.astype(float)
+            elif str(self._data.dtype) not in ["float64", "complex128"]:
```

Comment:
This is system dependend, isn't it?

Reply:
yes at least for the int. I'm not sure about float and complex. Should we do
`elif not str(self._data.dtype).startswith(("float", "complex")):`